### PR TITLE
Migrate `editing-widgets` to Playwright

### PR DESCRIPTION
**Pull request by kevin940726**
## What?

Part of #38851. Migrate `editing-widgets` to Playwright.

This is also the last missing piece to completely remove `puppeteer-testing-library` from the project.

## Why?

See #38851.

## How?

I discovered some bugs in the widgets screen during the migration. Also updated `insertBlock` and `getBlocks` to allow retrieving from a specified parent block.

## Testing Instructions

CI Should pass.

## Screenshots or screencast 
N/A

### CODE REVIEW

File: test/e2e/specs/widgets/editing-widgets.spec.js [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+			widgetsScreen.updateButton,
+			'Update button should start out disabled'
+		).toBeDisabled();
+
+		const [ firstWidgetArea, secondWidgetArea ] =
+			await widgetsScreen.widgetAreas.all();
+
+		await page
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
```
Comment (Mamaduka) [inline test/e2e/specs/widgets/editing-widgets.spec.js]: Nit (not a blocker): Should we make the locator more specific?

```js
await page
    .getByRole( 'toolbar', { name: 'Document tools' } )
    .getByRole( 'button', { name: 'Toggle block inserter' } )
    .click();
```

File: test/e2e/specs/widgets/editing-widgets.spec.js [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+			.filter( { hasText: 'Second Paragraph' } );
+
+		const secondParagraphBlockBoundingBox =
+			await secondParagraphBlock.boundingBox();
+
+		// Click outside the block to move the focus back to the widget area.
+		await firstWidgetArea.click( {
+			position: { x: 0, y: -1 },
+			// eslint-disable-next-line playwright/no-force-option
+			force: true,
```
Comment (Mamaduka) [inline test/e2e/specs/widgets/editing-widgets.spec.js]: Nit (not a blocker): It could be helpful in future to have reason for using `force: true`

**Review by Mamaduka — APPROVED** [review]
Thank you, @kevin940726!

I just had two minor notes, but migration looks great. I like that we could also enable a few skipped tests 🎉

> I discovered some bugs in the widgets screen during the migration.

Saw a few inline notes. Should we create separate issues for those?

File: test/e2e/specs/widgets/editing-widgets.spec.js (hunk 1 above)
Comment (kevin940726) [reply test/e2e/specs/widgets/editing-widgets.spec.js]: Sure!

File: test/e2e/specs/widgets/editing-widgets.spec.js (hunk 2 above)
Comment (kevin940726) [reply test/e2e/specs/widgets/editing-widgets.spec.js]: I thought I did 😅 .

Comment (kevin940726) [reply test/e2e/specs/widgets/editing-widgets.spec.js]: Turns out it's not doing what it seems, I refactored it 😅 .

**Pull request merged**